### PR TITLE
Add View Argument to Window.run()

### LINKED ISSUE #2400: Add optional `view` argument to `Window.run()' and `arcade.run()`
To help promote subclassing views over the window itself, we should add an optional view argument to the run methods, letting them automatically call `Window.show_view`.

It results in only a one-line change, but it makes the DX (Developer Experice) nicer.

It will also need a unit test for each method, and the examples will need updating to take advantage of the change.

### PR DESCRIPTION
**Pull request by EmmettBicker**
Fixes #2400 

I was looking at the Good First Issue tag and I found this one! Most of the work was already done in #2450 but that pr seems to have been abandoned, so I just applied the requested changes in here.

Please let me know if there's any issues :). This seems like a wonderful repo that I'd love to contribute to.

### CODE REVIEW

**Comment by einarf** [comment]
I was about to tag you in this issue, but you beat me :)

**Comment by EmmettBicker** [comment]
Haha, I was looking through the issues and it definitely seemed like a good place to start! Just joined the discord and am excited to contribute more

File: arcade/window_commands.py [hunk 1]
```diff
@@ -97,17 +97,23 @@ def close_window() -> None:
...
-    Run the main loop.
+    Run the main loop. Optionally start with a specified view.
 
     After the window has been set up, and the event hooks are in place,
     this is usually one of the last commands on the main program.
     This is a blocking function starting pyglet's event loop meaning
     it will start to dispatch events such as ``on_draw`` and ``on_update``.
+
+    Args:
+            view: The view to display when starting the run. Defaults to None.
```
Comment (einarf) [inline arcade/window_commands.py]: I'm not sure if sphinx-napoleon likes the additional 4 spaces here

File: arcade/window_commands.py [hunk 2]
```diff
@@ -97,17 +97,23 @@ def close_window() -> None:
     gc.collect()
 
 
-def run():
+def run(view: View | None = None):
```
Comment (einarf) [inline arcade/window_commands.py]: Might as well add a `None` return for this function while we are here

**Review by einarf — CHANGES_REQUESTED** [review]

File: arcade/window_commands.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 if TYPE_CHECKING:
     from arcade import Window
-
+    from arcade.application import View
```
Comment (einarf) [inline arcade/window_commands.py:19]: This one should ideally be further up with the other imports. Then do a `python make.py format` to ensure isort is happy.

**Comment by EmmettBicker** [comment]
On it! I only ran make.py lint and make.py test, I'll make sure to read the contributing doc more thoroughly

File: arcade/window_commands.py (hunk 3 above)
Comment (EmmettBicker) [reply arcade/window_commands.py:19]: When I tried this it caused a circular import error because application imports from window_commands

Comment (EmmettBicker) [reply arcade/window_commands.py:19]: Wait whoops that was the other file! I'll change this

Comment (EmmettBicker) [reply arcade/window_commands.py:19]: Oh actually I think it was this one

**Review by einarf — APPROVED** [review]

**Pull request merged**